pytorch_lightning/trainer/training_loop.py

```diff
@@ -494,6 +494,9 @@ def run_training_epoch(self):
             if batch_output.signal == -1:
                 break
 
+            # update LR schedulers
+            self.update_lr_schedulers('step')
```

Comment:
Dear @carmocca 
Exactly, thanks for pointing it out.
Will work around it further.

Reply:
Hi @carmocca , I realize that this could be a bit tricky. Because pl.module `on_train_batch_end` callbacks may log metrics, the LR scheduler must be updated following those callbacks. However, we must also make sure that a Checkpoint's `on_batch_batch_end` hook is called after the scheduler update.

Reply:
Hi @carmocca , in this case, the same issue may also exist for the validation part?
```python
        if not should_check_val or should_train_only:
            self.update_lr_schedulers('epoch')

        if should_train_only:
            self.check_checkpoint_callback(True)
```
if the metric is added in `on_validation_end` callbacks?

Reply:
Long story short, in that snippet you only update the epoch schedulers if running only training
and we had another call to update epoch schedulers if running validation too.
So no issue

However, this was just updated in #7357. It has no conflicts with this PR but you should rebase master regardless

Reply:
Hi @carmocca , it seems that this issue has been fixed by your updates!
https://colab.research.google.com/drive/1bBkhGiKJoavp1O4Oi4kLWVnxohl5HR6M?usp=sharing

Reply:
Oh! Sorry I didn't check myself.

Lovely when you unknowingly fix something :D